variation-commons-mongodb/src/test/java/uk/ac/ebi/eva/commons/mongodb/repositories/VariantRepositoryTest.java

```diff
@@ -99,42 +99,73 @@ public void checkFieldPresence() throws IOException {
 
     @Test
     public void testVariantIdIsFound() {
-        String id = "rs148957270";
+        List<String> ids = new ArrayList<>();
+        ids.add("rs148957270");
+
         List<VariantRepositoryFilter> filters = new ArrayList<>();
         List<String> exclude = new ArrayList<>();
         List<VariantMongo> variantEntityList = variantRepository
-                .findByIdsAndComplexFilters(id, filters, exclude, null);
+                .findByIdsAndComplexFilters(ids, filters, exclude, null);
         assertNotNull(variantEntityList);
         assertTrue(variantEntityList.size() > 0);
         Set<String> idSet = new HashSet<>();
-        idSet.add(id);
+        idSet.add(ids.get(0));
         idSet.add("ss254803838");
         assertEquals(idSet, variantEntityList.get(0).getIds());
     }
 
+    @Test
+    public void testVariantIdsFoundAndNotFound() {
+        List<String> ids = new ArrayList<>();
+
+        // Ids that should be found
+        ids.add("rs148957270");
+        ids.add("rs575961545");
+
+        // Id that should be not found
+        ids.add("a1b1c1d1e1");
+
+        List<VariantRepositoryFilter> filters = new ArrayList<>();
+        List<String> exclude = new ArrayList<>();
+        List<VariantMongo> variantEntityList = variantRepository
+                .findByIdsAndComplexFilters(ids, filters, exclude, null);
+        assertNotNull(variantEntityList);
+        assertTrue(variantEntityList.size() > 0);
+
+        // Only the 2 first ids should be found
+        assertEquals(2, variantEntityList.size());
```

Comment:
I would assert that the proper rs ids are found.

Reply:
@jorizci Maybe now? Or should I use another ids?

Reply:
Mmm not really happy with the assert, the important bit is that you are searching for the RS not checking the SS inside the variant. Also I don't remember if the query is returned always in the same order or not. My cautious side would prefer a iteration over the list and that the assert is that any variant that is returned on that list must contain either of those rs. 😦  I don't want to make it overly complicated. anyway.

Reply:
I agree with @jorizci , the asserts refer to IDs different from those in the request so it is a bit confusing.

Reply:
@cyenyxe @jorizci Yeah, sorry for that, now it checks that the RS are contained in one of the two lists (order independent), hope it's ok now :)